btf_types/src/lib.rs

```diff
@@ -136,8 +136,8 @@ impl BrainFuckProgram {
     }
 
     /// Get name of the file from where BF program is parsed.
-    pub fn filename(&self) -> String {
-        self.filename.clone()
+    pub fn filename(&self) -> &Path {
+        self.filename.as_path()
```

Comment:
While this is perfectly OK, you could also let the compiler do the deref for you and just return `&self.filename`